Documentation/design/neutron_plugin_for_mizar.md

```diff
@@ -0,0 +1,27 @@
+# Neutron plugin for mizar
+ 
+Openstack is one of the most popular opensource cloud platform, of which the network function subsystem is neutron. This Document is about to explan how to design a neutron plugin for mizar, in order that compute resources provided by openstack could use the mizar network pipeline. 
+
+neutron plugin might include two major functions: resource model save and transformation and network config distribution.
+
+## Resource model save and transformation
+
+The plugin should provide all operation APIs for mizar resources(vpc, network, endpoint, etc.), as the resource models of mizar are not same as neutron(router, network, subnet, port, etc.). Besides, neutron is one of the core subsystems of openstack, other core subsystems like nova may couple with neutron models. In order to make other subsystems of openstack like nova could properly interaction with neutron, when to create a mizar resource we also need to create one or more neutron resources corrspondingly. Those co-created neutron resources should only be stored, as the original neutron agents may not be needed any more and the mizar agents(transit agent, etc.) might only concern about the mizar resources.

```

Comment:
I'm not sure I fully understand the text. It will be helpful to provide examples and detailed workflows. This is how we design workflows in the new Mizar management plane: https://github.com/phudtran/mizar/blob/dev-k8s/Documentation/design/management_plane.md

Is this what you want to say?

```
1. The plugin shall transform all neutron's resource model's to Mizar resources
2. The Mizar resources shall be added to existing neutron network resources through API extension
3. The plugin shall program Mizar's transit daemon through RPC calls
4. ....
```